homeassistant/components/cover/isy994.py

```diff
@@ -0,0 +1,115 @@
+
+from homeassistant.components.cover import CoverDevice, DOMAIN
+from homeassistant.components.isy994 import (ISYDevice, NODES, PROGRAMS, ISY,
+                                             KEY_ACTIONS, KEY_STATUS,
+                                             filter_nodes)
+from homeassistant.const import STATE_OPEN, STATE_CLOSED, STATE_UNKNOWN
+from homeassistant.helpers.typing import ConfigType
+
+
+_LOGGER = logging.getLogger(__name__)
+
+VALUE_TO_STATE = {
+    0: STATE_CLOSED,
+    101: STATE_UNKNOWN,
+}
+
+UOM = ['97']
+STATES = [STATE_OPEN, STATE_CLOSED, 'closing', 'opening']
+
+
+def setup_platform(hass, config: ConfigType,
+                   add_devices: Callable[[list], None], discovery_info=None):
+    """Setup the ISY994 cover platform."""
+    if ISY is None or not ISY.connected:
+        _LOGGER.error('A connection has not been made to the ISY controller.')
+        return False
+
+    devices = []
+
+    for node in filter_nodes(NODES, units=UOM,
+                             states=STATES):
+        devices.append(ISYCoverDevice(node))
+
+    for program in PROGRAMS.get(DOMAIN, []):
+        try:
+            status = program[KEY_STATUS]
+            actions = program[KEY_ACTIONS]
+            assert actions.dtype == 'program', 'Not a program'
+        except (KeyError, AssertionError):
+            pass
+        else:
+            devices.append(ISYCoverProgram(program.name, status, actions))
+
+    add_devices(devices)
+
+
+class ISYCoverDevice(ISYDevice, CoverDevice):
+    """Representation of an ISY994 cover device."""
+
+    def __init__(self, node: object):
+        """Initialize the ISY994 cover device."""
+        ISYDevice.__init__(self, node)
+
+    @property
+    def current_cover_position(self) -> int:
+        """Get the current cover position."""
+        return sorted((0, self.value, 100))[1]
+
+    @property
+    def is_closed(self) -> bool:
+        """Get whether the ISY994 cover device is closed."""
+        return self.state == STATE_CLOSED
+
+    @property
+    def state(self) -> str:
+        """Get the state of the ISY994 cover device."""
+        return VALUE_TO_STATE.get(self.value, STATE_OPEN)
+
+    def open_cover(self, **kwargs) -> None:
+        """Send the open cover command to the ISY994 cover device."""
+        if not self._node.on(val=100):
+            _LOGGER.error('Unable to open the cover')
+
+    def close_cover(self, **kwargs) -> None:
+        """Send the close cover command to the ISY994 cover device."""
+        if not self._node.off():
+            _LOGGER.error('Unable to close the cover')
+
+
+class ISYCoverProgram(ISYCoverDevice):
+    """Representation of an ISY994 cover program."""
+
+    def __init__(self, name: str, node: object, actions: object) -> None:
+        """Initialize the ISY994 cover program."""
+        ISYCoverDevice.__init__(self, node)
+        self._name = name
+        self._actions = actions
+
+    @property
+    def is_closed(self) -> bool:
```

Comment:
No need to overwrite this if you change your logic in `state` to use `self.value`